### PR TITLE
Fix `stylelint.utils.checkAgainstRule()` regression when both `reportNeedlessDisables` and `quiet` enabled

### LINKED ISSUE #7843: Valid disables reported when `--report-needless-disables` and `--quiet` are set
### What minimal example or steps are needed to reproduce the bug?

- Add a custom plugin using `stylelint.utils.checkAgainstRule()`

```js
import stylelint from "stylelint";

const ruleName = "custom/plugin";

function rule(primary, secondary, context) {
  return async (root, result) => {
    await stylelint.utils.checkAgainstRule(
      {
        ruleName: "block-no-empty",
        ruleSettings: [primary, secondary],
        root,
        result,
        context
      },
      (warning) => {
        stylelint.utils.report({
          ruleName,
          result,
          message: warning.text,
          node: warning.node,
        });
      }
    );
  };
};

export default stylelint.createPlugin(ruleName, rule);
```

- Add an `example.css` file with the `checkAgainstRule` disabled

```css
/* stylelint-disable */
a {}
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
  plugins: ["./plugins/custom-plugin"],
  rules: { "custom/plugin": true },
};
```

### How did you run Stylelint?

```shell
stylelint example.css --report-needless-disables
```
```shell
stylelint example.css --report-needless-disables --quiet
```

### Which Stylelint-related dependencies are you using?

```json
{
  "devDependencies": {
    "stylelint": "16.7.0"
  }
}
```

### What did you expect to happen?

Expected both runs of Stylelint to report no errors

### What actually happened?

- ✅ Stylelint with `--report-needless-disables` option set reports no errors
- ❌ Stylelint with `--report-needless-disables` and `--quiet` options set reports errors

image

### Do you have a proposal to fix the bug?

I don't have a proposal. However, I did link `stylelint` to my project, stepped back through the history, and identified that the failures started with this [PR](https://github.com/stylelint/stylelint/pull/7833). Additionally, I began reverting changes line by line and pinpointed the issue to [this change in checkAgainstRule](https://github.com/stylelint/stylelint/pull/7833/files#diff-7d9680e3687d5b028c2a7a394026ed850a4d45b704584a2d9001c86d32006d34R46).

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Closes #7843
Related to #7833

> Is there anything in the PR that needs further explanation?

This PR also adds constants about the severities (`error`/`warning`). In the follow-up PRs, we can refactor to replace string literals with the constants.

### CODE REVIEW

**Review by aaronccasanova — APPROVED** [review]
🚀

File: lib/constants.mjs [hunk 1]
```diff
@@ -15,3 +15,7 @@ export const EXIT_CODE_INVALID_USAGE = 64;
 export const EXIT_CODE_INVALID_CONFIG = 78;
 
 export const RULE_NAME_ALL = 'all';
+
+export const SEVERITY_ERROR = 'error';
```
Comment (Mouvedia) [inline lib/constants.mjs:19]: https://github.com/stylelint/stylelint/blob/e5a2436cc146de1d97f7d87c03dcd66ba8cc8cd2/lib/reportDisables.mjs#L31
https://github.com/stylelint/stylelint/blob/e5a2436cc146de1d97f7d87c03dcd66ba8cc8cd2/lib/utils/reportCommentProblem.mjs#L32-L38
https://github.com/stylelint/stylelint/blob/e5a2436cc146de1d97f7d87c03dcd66ba8cc8cd2/lib/reportUnknownRuleNames.mjs#L70
https://github.com/stylelint/stylelint/blob/e5a2436cc146de1d97f7d87c03dcd66ba8cc8cd2/lib/validateDisableSettings.mjs#L63
https://github.com/stylelint/stylelint/blob/e5a2436cc146de1d97f7d87c03dcd66ba8cc8cd2/lib/formatters/preprocessWarnings.mjs#L36
https://github.com/stylelint/stylelint/blob/e5a2436cc146de1d97f7d87c03dcd66ba8cc8cd2/lib/formatters/preprocessWarnings.mjs#L48
https://github.com/stylelint/stylelint/blob/e5a2436cc146de1d97f7d87c03dcd66ba8cc8cd2/lib/createPartialStylelintResult.mjs#L95
https://github.com/stylelint/stylelint/blob/e5a2436cc146de1d97f7d87c03dcd66ba8cc8cd2/lib/lintPostcssResult.mjs#L90

File: .changeset/stale-geckos-own.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: `stylelint.utils.checkAgainstRule()` to prevent false problems when both `reportNeedlessDisables` and `quiet` enabled
```
Comment (Mouvedia) [inline .changeset/stale-geckos-own.md:5]:
```suggestion
Fixed: `stylelint.utils.checkAgainstRule()` regression when `reportNeedlessDisables` and `quiet` are both enabled
```

**Review by romainmenke — APPROVED** [review]
LGTM! Thank you

**Pull request merged**